sdk/zklogin/src/utils.ts

```diff
@@ -15,6 +15,27 @@ export function getExtendedEphemeralPublicKey(publicKey: PublicKey) {
 	return toB64(publicKey.toSuiBytes());
 }
 
+export function toBigIntBE(bytes: Uint8Array) {
+	const hex = toHEX(bytes);
+	if (hex.length === 0) {
+		return BigInt(0);
+	}
+	return BigInt(`0x${hex}`);
+}
+
+/** Takes either a bigint, a string-encoded bigint, or a string-encoded Base64 value, and returns a bigint */
+export function unknownToBigInt(value: bigint | string) {
+	if (typeof value === 'bigint') {
+		return value;
+	}
+
+	if (/^\d+$/.test(value)) {
```

Comment:
this seems linda sketchy, what's the situation where we don't know if we are getting a decimal string vs a base64 string?  A lot of decimal numbers can also be interpreted as base64... 

Is the assumption here that any integer thats base 64 encoded will include non-decimal characters?

Reply:
Yeah so technically I don't need to do this, and could keep it as base64 bigint | string, but there's some APIs that we use hand us bigint strings, and it'd maybe be nice to pipe those through without issue. Right now if you do that you might get an error, or maybe it'll _look_ like it works because there's no typeerror, but actually fail.

Reply:
I think for randomness it probably doesn't matter, but this has the potential for a horrible thing to debug.

I don't know if its possible, but if 99% of the time you pass in base64, and it decodes it correctly, but then a base64 value happens to be all decimal characters, and it decodes the wrong thing and breaks your app in some way, reproducing that is going to be basically impossible

Reply:
Yeah this is good callout. Let me set some time up to sync with you here too